### PR TITLE
Added VO Cone Search tutorial as per astropy/astropy-tutorials#15

### PR DESCRIPTION
**Pull request by pllim**
Added VO Cone Search tutorial iPython notebook, as requested by astropy/astropy-tutorials#15 .

[Edit: here's a link to the rendered notebook http://nbviewer.ipython.org/github/pllim/astropy-tutorials/blob/vo-conesearch-tutorial/tutorials/vo/conesearch_tutorial.ipynb]

### CODE REVIEW

**Comment by pllim** [comment]
@adrn , based on a reply from @eteq , I am going to wait till after v1.0 release. FYI.

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# Import conesearch\n",
+      "from astropy.vo.client import conesearch"
```
Comment (pllim) [inline tutorials/vo/conesearch_tutorial.ipynb]: Done at the top of tutorial.

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Let's predict the run time of performing Cone Search on `http://gsss.stsci.edu/webservices/vo/ConeSearch.aspx?CAT=GSC23&` with a radius of 0.5 degrees. For now, the prediction assumes a very simple linear model, which might or might not reflect the actual trend.\n",
```
Comment (pllim) [inline tutorials/vo/conesearch_tutorial.ipynb]: Done.

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "What if you want *all* the results from *all* the catalogs? And you also want to suppress all the VO table warnings and informational messages?\n",
```
Comment (pllim) [inline tutorials/vo/conesearch_tutorial.ipynb]: Done.

**Comment by pllim** [comment]
@adrn , tutorial is updated.

**Comment by adrn** [comment]
Thanks @pllim -- reviewing this now

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Let's explore the 3 rows of astronomical objects found within 0.1 degree of M31 in the given catalog and sort them by increasing distance. For this example, the VO table has 4 columns:\n",
+      "\n",
+      "* `_r` = Distance of object, in degrees, from M31\n",
```
Comment (adrn) [inline tutorials/vo/conesearch_tutorial.ipynb]: Just to make this clear, maybe say "Angular distance (in degrees) between object and M31" or something like that

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+     "source": [
+      "If you have a favorite catalog in mind, you can also perform Cone Search only on that catalog."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "result = conesearch.conesearch(\n",
+      "    c, 0.1 * u.degree, catalog_db='The USNO-A2.0 Catalogue (Monet+ 1998) 1')"
```
Comment (adrn) [inline tutorials/vo/conesearch_tutorial.ipynb]: Could you describe or link to documentation here that specifies what are valid inputs for `catalog_db` (e.g., where can I find the valid catalog names)

File: tutorials/vo/conesearch_tutorial.ipynb [hunk 6]
```diff
@@ -0,0 +1,438 @@
...
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "import numpy as np\n",
```
Comment (adrn) [inline tutorials/vo/conesearch_tutorial.ipynb]: Move this and all other imports to the top of the notebook

**Comment by adrn** [comment]
@pllim First off, thanks for doing this -- it looks good, but I think we could add some things that will make it an even better tutorial. I would say that one of the main goals of the tutorials is to show how the different astropy subpackages interact and work together. Right now, this reads a bit like documentation and is primarily just focused on conesearch (and some Quantity). I don't think it would be much work to add a few things to highlight synergies and make it feel more like a workflow. 

A few ideas (you don't have to do all or any of these, just some ideas for how to expand the tutorial a bit):

 * Add a section where you write out a catalog to disk using `astropy.table`
 * Query a catalog and turn it into a Latex table
 * Use aplpy and overplot circles marking known sources with some constraint over an image
 * Use an SDSS query to get positions for sources, then query the SDSS SAS (http://dr12.sdss3.org/) to get back a few spectra to plot

Of course, if you have your own ideas, that works too -- thanks!

cc @kelle @eteq

**Comment by pllim** [comment]
@adrn , thanks for the ideas. Are those usually how people use Cone Search results? I would like the tutorial to be realistic rather than adding things just for the sake of showcasing `astropy`. Also, maybe adding the extra stuff can be a separate PR? This one has been open for over a year.

I'll be glad to do the small changes mentioned within this PR though.

File: tutorials/vo/conesearch_tutorial.ipynb (hunk 5 above)
Comment (pllim) [inline tutorials/vo/conesearch_tutorial.ipynb]: I clarified in the tutorial that the catalog names can be obtained using `list_catalog()` in an earlier entry.

**Comment by adrn** [comment]
@pllim sure, we can add this in a new PR -- I'll create an issue about this, but I do think it would be useful (and realistic) to chose implement the things I mention above.

**Comment by pllim** [comment]
Thanks. I see that you have opened #82.

**Pull request merged**